docs/src/modules/components/Link.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable jsx-a11y/anchor-has-content */
+import * as React from 'react';
+import clsx from 'clsx';
+import { useRouter } from 'next/router';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import { useUserLanguage } from 'docs/src/modules/utils/i18n';
+
+type NextLinkComposedProps = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> &
+  Omit<NextLinkProps, 'href' | 'as'> & {
+    to: NextLinkProps['href'];
+    linkAs?: NextLinkProps['as'];
+    href?: any;
+  };
+
+export const NextLinkComposed = React.forwardRef<HTMLAnchorElement, NextLinkComposedProps>(
+  (props, ref) => {
+    const { to, linkAs, href, replace, scroll, passHref, shallow, prefetch, ...other } = props;
+
+    return (
+      <NextLink
+        href={to}
+        prefetch={prefetch}
+        as={linkAs}
+        replace={replace}
+        scroll={scroll}
+        shallow={shallow}
+        passHref={passHref}
+      >
+        <a ref={ref} {...other} />
+      </NextLink>
+    );
+  },
+);
+
+export type LinkProps = {
+  activeClassName?: string;
+  as?: NextLinkProps['as'];
+  noLinkStyle?: boolean;
+} & Omit<NextLinkComposedProps, 'to' | 'linkAs'> &
+  Omit<MuiLinkProps, 'href'>;
+
+// A styled version of the Next.js Link component:
+// https://nextjs.org/docs/#with-link
+const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(function Link(props, ref) {
+  const {
+    activeClassName = 'active',
+    as: linkAsProp,
+    className: classNameProps,
+    href,
+    noLinkStyle,
+    role: roleProp,
+    ...other
+  } = props;
+
+  // Catch role passed from ButtonBase. This is definitely a link
+  // @ts-ignore
```

Comment:
```suggestion
  // @ts-expect-error
```
And explain why.

Will look into forbidding `@ts-ignore` via linter. I'm not aware of any reason to keep using `@ts-ignore` in unpublished types.